python/cugraph/community/ecg_wrapper.pyx

```diff
@@ -61,10 +62,9 @@ def ecg(input_graph, min_weight=.05, ensemble_size=16):
     g.adjList.get_vertex_identifiers(&c_index_col)
 
     df['partition'] = cudf.Series(np.zeros(num_verts, dtype=np.int32))
-    #cdef uintptr_t c_ecg_ptr = get_column_data_ptr(df['partition']._column)
     cdef uintptr_t c_ecg_ptr = df['partition'].__cuda_array_interface__['data'][0]
 
-    if g.adjList.edge_data.dtype == np.float32:
+    if weights.dtype == np.float32:
```

Comment:
Does this work if weights is None?

Reply:
Obviously not :-)

Pushed another more complex change.  In fact the implementation requires a weighted graph, so the new version raises an exception on the python side if there are no weights.